packages/wallets/strike/src/adapter.ts

```diff
@@ -0,0 +1,181 @@
+
+    private _connecting: boolean;
+    private _wallet: StrikeWallet | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+
+
+    constructor(config: StrikeWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            scopePollingDetectionStrategy(() => {
+                if (this._wallet) {
+                    this._readyState = WalletReadyState.Loadable;
+                    this.emit('readyStateChange', this._readyState);
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            let StrikeClass: typeof StrikeWallet;
+            let wallet: StrikeWallet;
+            await import('@strike-protocols/solana-wallet-adapter').then(async (strikeSolanaWalletAdapter) => {
```

Comment:
`await...then(async ...)` is an antipattern. We're already in an async function, each promise can be awaited in order and without creating an additional closure. Take a look at the Torus adapter, it does this quite simply.

Reply:
thanks, we started with the torus pattern but typescript wasn't happy with it... will give it another shot

Reply:
ok, works now, not sure what the problem was before